tests/unit/test_models/test_base_model.py

```diff
@@ -922,6 +925,61 @@ def test_set_initial_condition_errors(self):
         with self.assertRaisesRegex(pybamm.ModelError, "must appear in the solution"):
             model.set_initial_conditions_from({"wrong var": 2})
 
+    def test_latexify(self):
+        model_dfn = pybamm.lithium_ion.DFN()
+        func_dfn = str(model_dfn.latexify())
+
+        model_spme = pybamm.lithium_ion.SPMe()
+        func_spme = str(model_spme.latexify())
+
+        # Test newline=False
+        self.assertIn(r"\\", str(model_spme.latexify(newline=False)))
+
+        # Test terminal voltage equation name
+        self.assertIn("Terminal voltage [V]", func_spme)
+
+        # Test partial derivative in boundary conditions
+        self.assertIn("partial r}", func_spme)
+
+        # Test boundary conditions range
+        self.assertIn("quad r =", func_spme)
+
+        # Test derivative in equations
+        self.assertIn("frac{d}{d t}", func_spme)
+
+        # Test rhs geometry ranges
+        self.assertIn("quad 0 < r < 1", func_spme)
+
+        # Test initial conditions
+        self.assertIn("; t=0", func_spme)
+
+        # Test DFN algebraic lhs
+        self.assertIn("0 =", func_dfn)
+
+        # Test concatenation cases
+        try:
+            self.assertIn("begin{cases}", func_spme)
+            self.assertIn("end{cases}", func_spme)
+        except AssertionError:
+            for eqn in model_spme.rhs.values():
+                concat_displays = model_spme._get_concat_displays(eqn)
+                if concat_displays:
+                    self.assertIn("begin{cases}", str(concat_displays))
+                    self.assertIn("end{cases}", str(concat_displays))
+                    break
+
+        # Test sympy preview (skip workflow for mac, windows)
+        if platform.system() == "Linux":
+            for ext in ["png", "jpg", "pdf"]:
+                filename = f"{uuid.uuid4()}.{ext}"
+                model_spme.latexify(filename)
+                os.remove(filename)
```

Comment:
This could be a separate test skipped with the decorator `skipIf` (to make it obvious it is skipped for mac and windows)